FrontEnd/src/components/Browse.tsx

```diff
@@ -1,10 +1,176 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import {
+  TableContainer,
+  Table,
+  TableHead,
+  TableBody,
+  TableRow,
+  TableCell,
+  Grid,
+  Button,
+} from '@mui/material';
+import DownloadIcon from '@mui/icons-material/Download';
+import DeleteIcon from '@mui/icons-material/Delete';
+import EditIcon from '@mui/icons-material/Edit';
+import data from '../mock-table-data.json';
 
 const Browse = () => {
+  let handleRemove = (i: any) => {
+    const newRows = [...rows];
+    const index = rows.findIndex((row) => row.file === i);
+    newRows.splice(index, 1);
+    setRows(newRows);
+  };
+
+  //PERMANENT CODE BELOW
+  //Commented out code is for use once the backend is set up.
+  //Once set up, simply replace fetch call with correct link, uncomment the code,
+  //and then delete the TEMPORARY code at the end of the file.
```

Comment:
We should avoid keeping two versions of the code here because the commented out one will become outdated and then it will be hard to use it to replace the existing code.

Instead, you should use the error/isLoaded state into existing code.

Instead of passing the data directly like this:

```
const [rows, setRows] = useState(data);
```

You should make an async function that returns the data to simulate the server call at the top of the file:

```javascript
async function getData() {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(data);
    }, 2000);
  });
}
```

And then use it as if that was the `fetch('....')` call:

```
getData().then((data) => setRows(data))
````

And that way you can use the isLoaded variable too to show a "Loading" text etc. without having to keep temporary code here.

Reply:
And that allows us to develop and test the loading code & the error code as well